colink/protocol.py

```diff
@@ -30,9 +31,14 @@ def decorator(func):
 
         return decorator
 
-    def run(self, cl: CoLink = None, keep_alive_when_disconnect: bool = False):
+    def run(
+        self,
+        cl: CoLink = None,
+        keep_alive_when_disconnect: bool = False,
+        vt_public_addr: str = "",
```

Comment:
I checked again and I think that because when reading the environment variables, if not provided I will set vt_public_addr as empty string. So later I decide to use empty string to represent this state.

Reply:
Yeah in that case, maybe for the env var we should also set to None when not provided?

Reply:
In fact, why are the parsing default for all args `""` rather than `None`? I don't think it is a good practice. When you write the downstream code, it is hard to figure out whether something is not provided, or is provided but is set to an empty string. I cannot think of a single reason to use string here, and also for other args.